WT-10036 Touch-ups to wtperf_run.sh

ajmorton opened the pull request:
Originally this ticket was to create a new script to get the average results of multiple wtperf test runs, but `wtperf_run.sh` achieves this goal already so I've made a few small tweaks to this script instead. Namely:
- Ran shellcheck on the script and applied recommended changes
- Update traces and save final results to file
- Add code to report the number of checkpoints in wtperf tests, as this is a metric we are tracking in some of our `evergreen.yml` perf tests.

@sueloverso I assume this script is no longer used as of the migration from Jenkins to Evergreen, but just in case please let me know changing the name of `wtperf.out` breaks any existing workflows.

ajmorton commented on bench/wtperf/runners/wtperf_run.sh:
```diff
@@ -34,36 +34,38 @@ while [[ $# -gt 0 ]] ; do
 getval()
 {
-	max="$1"
+	get_max="$1"
```
shellcheck: this was shadowing the global array `max` declared on old line 43

sueloverso reviewed (commented):
These changes themselves look fine. I think the script also needs a pass to update the comments. There are several comments that refer to Jenkins, for example, and one that states we do 5 runs (even though that's a passed in arg). Feel free to add more comments for any lines that weren't obvious what they were doing.

ajmorton commented:
Thanks @sueloverso!
I've updated comments, and also changed up the traces to look like the following:
```
small-btree.wtperf: runmax=5, wtarg=''
Run 1:
        read:   29126108
        load:   0.21
Run 2:
        read:   28814339
        load:   0.20
...
Average:
        Read count:      28892694
        Load time:       .203
```

Our current performance tests are now handled by the `run-perf-test` task in evergreen.yml, which uses  `bench/perf_run_py/perf_run.py`. The question above was in case we have some non-evergreen workflows I wasn't aware of but it seems we're ok on that front.

lukech commented:
@sueloverso 
> I don't know where the performance tests are currently run and maintained. Are the performance tests using this script or not?

As pointed out by Andrew in the last comment, perf tests run in the "Ubuntu 20.04 Performance tests" Evergreen build variant, using function `run-perf-test`, which calls `bench/perf_run_py/perf_run.py` wrapper. I did not notice `wtperf_run.sh` wrapper is actively used by any Evergreen task, and assume it's a script people run locally.

clarissecheah commented on bench/wtperf/runners/wtperf_run.sh:
```diff
@@ -1,16 +1,23 @@
+# Usage:
+# 	wtperf_run.sh test_conf numruns [wtperf_args]
+# 		test_conf:   The path the the wtperf configuration
+# 		rumruns:     How many times to run the test
```
Is this meant to be numruns? Also, to remain consistent, could we change that to num_runs?

ajmorton replied:
Yes it is 👍 
I ended up changing the names of the other two arguments so that they match the variable names used in the script. Updated in d5ee691

ajmorton commented:
@clarissecheah has found a much better way to do this via `perf_run.sh` which is the code we currently use to run our perf tests in `evergreen.yml`. Given we won't be using `wtperf_run.sh` I'll close off this PR.
